lib/streamlit/server/Server.py

```diff
@@ -116,8 +118,35 @@ def start(self, on_started):
 
         LOGGER.debug("Starting server...")
         app = self._create_app()
+
+        class PortAlradyInUse(Exception):
+            pass
+
+        def start_listening(rotate=True):
+            port = config.get_option("server.port")
+            try:
+                app.listen(port)
+            except OSError as e:
+                ADDRESS_ALREADY_IN_USE = 48
+                if e.errno == ADDRESS_ALREADY_IN_USE:
+                    if config.is_manually_set("server.port"):
+                        LOGGER.debug(
+                            "Port %s already in use, trying next available one", port
+                        )
+                        port += 1
+                        if port == 3000: port += 1 # save the 3000
```

Comment:
Why not set it to 3000? Maybe update the comment

Reply:
@tvst what is the reason to save this port from being used? it looks like a "just in case" situation

Reply:
i'm think it's being used by the frontend server

Reply:
3000 is treated as a special port in the frontend code. When the port is 3000 we assume the websocket should be connected to 8501 (served from Server.py) rather than 3000 (served from the dev server).

We should change that logic, but without changing it the best solution is to skip 3000 here.

Also: please add a comment here explaining to future generations what I just said :)